### PR TITLE
{lib}[fosscuda/2018a] TensorFlow v1.8.0

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)
TensorFlow 1.8.0 with fosscuda/2018a

### CODE REVIEW

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 2 out of 4 (4 easyconfigs in this PR)
nic168 - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, Python 2.7.5
See https://gist.github.com/afbd4a6a22f2d76b24c658aa69d11959 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 10 out of 12 (4 easyconfigs in this PR)
dell.lan - Linux fedora 28, Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz, Python 2.7.15
See https://gist.github.com/c1066671a5aea4352c05b11549492abe for a full test report.

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.8.0-fosscuda-2018a-Python-3.6.4.eb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+toolchain = {'name': 'fosscuda', 'version': '2018a'}
+
+source_urls = ['https://github.com/tensorflow/tensorflow/archive/']
+sources = ['v%(version)s.tar.gz']
+patches = [
+    'TensorFlow-1.5.0_swig-env.patch',
+    'TensorFlow-%(version)s_remove-msse-hardcoding.patch',
+    'TensorFlow-%(version)s_lrt-flag.patch',
+    'TensorFlow-1.6.0_use_64b_alignment.patch',
+    'TensorFlow-1.6.0_dont_expand_cudnn_path.patch',
```
Comment (wpoely86) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.8.0-fosscuda-2018a-Python-3.6.4.eb]: @akesandgren this one is missing

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.8.0-fosscuda-2018a-Python-3.6.4.eb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    'TensorFlow-%(version)s_use_mklddn_0.14.patch',
+    'TensorFlow-%(version)s_GetVersion_tuple_2_struct.patch',
+]
+checksums = [
+    '47646952590fd213b747247e6870d89bb4a368a95ae3561513d6c76e44f92a75',  # v1.8.0.tar.gz
+    '53807290f1acb6a0f2a84f1a0ad9f917ee131c304b3e08679f3cbd335b22c7ef',  # TensorFlow-1.5.0_swig-env.patch
+    '4d043bba84d19aa38e402ef4ddaa89df936ae792f82693a49ce8ae4a1d091f6d',  # TensorFlow-1.8.0_remove-msse-hardcoding.patch
+    '3c499788266800c71fa9874d411df79f7331320b3a86bc7c79c74763ae04c143',  # TensorFlow-1.8.0_lrt-flag.patch
+    'bcff66b5cc7a98e2ae894df6940aac5e3dcec3700773d24a6c3a5a69128ef66f',  # TensorFlow-1.6.0_use_64b_alignment.patch
+    '37ba1f77f42252b7484b32b8309f9acf1f910612623581a98fd39c9f94e9b285',  # TensorFlow-1.8.0_dont_expand_cuda_cudnn_path.patch
```
Comment (wpoely86) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.8.0-fosscuda-2018a-Python-3.6.4.eb]: you have some lines here past the 120 limit.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 4 (4 easyconfigs in this PR)
nic168 - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, Python 2.7.5
See https://gist.github.com/00a7dc92a3e3abca9102d17ee29ac0f2 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 4 (4 easyconfigs in this PR)
nic167 - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, Python 2.7.5
See https://gist.github.com/9c2ab7c991a75e80f669647c5f8d557a for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 10 out of 12 (4 easyconfigs in this PR)
easybuild.novalocal - Linux centos linux 7.5.1804, Intel Xeon E312xx (Sandy Bridge), Python 2.7.5
See https://gist.github.com/55a5612335c5ec13a3756f422b88d9c0 for a full test report.

**Comment by verdurin** [comment]
@akesandgren As currently available the relevant tarball is:
`cudnn-9.1-linux-x64-v7.tgz` so you either need to leave a comment asking for it to be renamed or alter the version handling.

**Comment by akesandgren** [comment]
@verdurin See the comment in the cudnn---.eb file :-)

**Comment by verdurin** [comment]
@akesandgren It's not exactly crystal clear from that comment that the user has to rename the downloaded file...

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 1 out of 2 (4 easyconfigs in this PR)
easybuild.novalocal - Linux centos linux 7.5.1804, Intel Xeon E312xx (Sandy Bridge), Python 2.7.5
See https://gist.github.com/3f8607129f6c1392c473535f5df61733 for a full test report.

**Comment by boegel** [comment]
@akesandgren Does this work for you in its current state (and can you submit a test report)?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 1 out of 4 (4 easyconfigs in this PR)
nic167 - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, Python 2.7.5
See https://gist.github.com/3f0b7ae753e05ee734b321f5c082d627 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 0 out of 0 (4 easyconfigs in this PR)
b-an04.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/33f0e99c02f2d8f94bc22ba6e03f3ace for a full test report.

**Comment by boegel** [comment]
@verdurin @wpoely86 Any idea why the test reports are failing for you?

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
b-an04.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/87719383bd8554c8bd285f243a34b3f1 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (4 easyconfigs in this PR)
easybuild.novalocal - Linux centos linux 7.5.1804, Intel Xeon E312xx (Sandy Bridge), Python 2.7.5
See https://gist.github.com/7139e0e33446871c43041209172e5139 for a full test report.

**Comment by verdurin** [comment]
@boegel this is right at the end of the build error log:
```
bazel-out/host/bin/tensorflow/cc/ops/data_flow_ops_gen_cc: error while loading shared libraries: libcuda.so.1: cannot open shared object file: No such file or directory
Target //tensorflow/tools/pip_package:build_pip_package failed to build
```

**Comment by verdurin** [comment]
Actually there is no such file, while there is a `libcudart.so.9.1` provided by the module...

**Comment by akesandgren** [comment]
libcuda.so.1 comes from the libcuda1-3xx OS package.
It's not part of the CUDA EB package.

**Comment by verdurin** [comment]
@akesandgren if that's required, then it should be listed as an OS dep.

**Comment by akesandgren** [comment]
I wish... but you can't do that for one very simple reason.
At least on Debian/Ubuntu, the nvidia kernel driver and libcuda1 package are tightly coupled (as i guess they are on RH/CentOS too).
On Ubuntu, they are called nvidia-373, nvidia-386, nvidia-391, nvidia-396 and so on depending on which version of the driver it is. So such a OS dep list would in theory be infinitely long. Or at least way too long and undeterminate.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
nic169 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, Python 2.7.5
See https://gist.github.com/a13567d52cd60de95c6a85f66d42e88c for a full test report.

**Comment by boegel** [comment]
So, is this good to go? @verdurin?

File: easybuild/easyconfigs/p/Python/Python-3.6.4-fosscuda-2018a.eb [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+# python needs bzip2 to build the bz2 package
+dependencies = [
+    ('bzip2', '1.0.6'),
+    ('zlib', '1.2.11'),
+    ('libreadline', '7.0'),
+    ('ncurses', '6.0'),
+    ('SQLite', '3.21.0'),
+    ('GMP', '6.1.2'),  # required for pycrypto
+    ('libffi', '3.2.1'),  # required for cryptography
```
Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-3.6.4-fosscuda-2018a.eb]: @akesandgren missing `XZ` dependency...

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.8.0-fosscuda-2018a-Python-3.6.4.eb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    ('Bazel', '0.12.0'),
+    ('wheel', '0.31.0', versionsuffix),
+]
+
+dependencies = [
+    ('Python', '3.6.4'),
+    ('cuDNN', '7.0.5.15'),
+]
+
+preconfigopts = 'export CC_OPT_FLAGS="$CC_OPT_FLAGS -mno-avx512f" && '
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.8.0-fosscuda-2018a-Python-3.6.4.eb]: @akesandgren Could this explain https://github.com/easybuilders/easybuild-easyconfigs/issues/6515?

Comment (akesandgren) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.8.0-fosscuda-2018a-Python-3.6.4.eb]: No, it's the GPU part that is slower.
Actually the upstream wheel is still built without AVX2 AVX512F FMA.

File: easybuild/easyconfigs/p/Python/Python-3.6.4-fosscuda-2018a.eb (hunk 3 above)
Comment (akesandgren) [reply easybuild/easyconfigs/p/Python/Python-3.6.4-fosscuda-2018a.eb]: Odd, it's there in the .eb file I'm using to actually build...

**Comment by boegel** [comment]
@akesandgren Another test report after the last changes would be nice.

Do you consider https://github.com/easybuilders/easybuild-easyconfigs/issues/6515 to be a blocker for this?

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
b-cn1522.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) Gold 6132 CPU @ 2.60GHz, Python 2.7.12
See https://gist.github.com/2ef2e02ff9b62999a95f05b1f149537b for a full test report.

**Comment by boegel** [comment]
Going in, thanks @akesandgren!

**Pull request merged**